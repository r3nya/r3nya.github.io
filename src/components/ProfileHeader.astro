---
interface Props {
  name: string;
  headline: string;
}

const { name, headline }: Props = Astro.props;
---

<header class="col-span-1 md:col-span-2">
  <h1 class="text-5xl font-bold mb-4 md:text-6xl mt-2">{name}</h1>
  <div
    class="mb-4 h-1 rounded-full bg-gradient-to-r from-gradient-from-light via-gradient-via-light to-gradient-to-light dark:from-gradient-from-dark dark:via-gradient-via-dark dark:to-gradient-to-dark"
  >
  </div>
  <h2
    class="text-2xl font-bold text-palette-3-light md:text-3xl xl:text-4xl dark:text-palette-3-dark"
  >
    {headline}
  </h2>
</header>