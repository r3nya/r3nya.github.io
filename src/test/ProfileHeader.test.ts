// @vitest-environment happy-dom
import { describe, it, expect } from 'vitest';
import ProfileHeader from '../components/ProfileHeader.astro';
import { renderAstroComponent } from './helpers';

describe('ProfileHeader', () => {
  it('renders with English props', async () => {
    const result = await renderAstroComponent(ProfileHeader, {
      props: {
        name: 'Andrey Makarov',
        headline: 'Software Developer 👨‍💻',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Russian props', async () => {
    const result = await renderAstroComponent(ProfileHeader, {
      props: {
        name: 'Андрей Макаров',
        headline: 'Разработчик ПО 👨‍💻',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });

  it('renders with Spanish props', async () => {
    const result = await renderAstroComponent(ProfileHeader, {
      props: {
        name: 'Andrey Makárov',
        headline: 'Desarrollador 👨‍💻',
      },
    });

    expect(result.innerHTML).toMatchSnapshot();
  });
});
