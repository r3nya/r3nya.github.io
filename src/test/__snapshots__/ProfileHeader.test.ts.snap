// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProfileHeader > renders with English props 1`] = `"<header class="col-span-1 md:col-span-2"> <h1 class="text-5xl font-bold mb-4 md:text-6xl mt-2">Andrey Makarov</h1> <div class="mb-4 h-1 rounded-full bg-gradient-to-r from-gradient-from-light via-gradient-via-light to-gradient-to-light dark:from-gradient-from-dark dark:via-gradient-via-dark dark:to-gradient-to-dark"></div> <h2 class="text-2xl font-bold text-palette-3-light md:text-3xl xl:text-4xl dark:text-palette-3-dark"> Software Developer 👨‍💻 </h2> </header>"`;

exports[`ProfileHeader > renders with Russian props 1`] = `"<header class="col-span-1 md:col-span-2"> <h1 class="text-5xl font-bold mb-4 md:text-6xl mt-2">Андрей Макаров</h1> <div class="mb-4 h-1 rounded-full bg-gradient-to-r from-gradient-from-light via-gradient-via-light to-gradient-to-light dark:from-gradient-from-dark dark:via-gradient-via-dark dark:to-gradient-to-dark"></div> <h2 class="text-2xl font-bold text-palette-3-light md:text-3xl xl:text-4xl dark:text-palette-3-dark"> Разработчик ПО 👨‍💻 </h2> </header>"`;

exports[`ProfileHeader > renders with Spanish props 1`] = `"<header class="col-span-1 md:col-span-2"> <h1 class="text-5xl font-bold mb-4 md:text-6xl mt-2">Andrey Makárov</h1> <div class="mb-4 h-1 rounded-full bg-gradient-to-r from-gradient-from-light via-gradient-via-light to-gradient-to-light dark:from-gradient-from-dark dark:via-gradient-via-dark dark:to-gradient-to-dark"></div> <h2 class="text-2xl font-bold text-palette-3-light md:text-3xl xl:text-4xl dark:text-palette-3-dark"> Desarrollador 👨‍💻 </h2> </header>"`;
